src/Repository/ForumCategoryRepository.php

```diff
@@ -29,4 +29,25 @@ public function findPaginated(int $page, int $maxPerPage = 25): Pagerfanta {
 
         return $pager;
     }
+
+    public function findCategories($isAdmin, $modForumId) {
```

Comment:
If anyone can help rewrite this to be more efficient please feel free!

Reply:
Add this as a comment above the method

Reply:
Feels like this whole filtering can be done on SQL level.
Create a task and assign to anyone who mentioned "ORM being inefficient" lol

Reply:
Added it as a TODO in comment. Will add task after PR get's merged.